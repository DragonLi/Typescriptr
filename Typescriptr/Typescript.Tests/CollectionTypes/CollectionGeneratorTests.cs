using System.Collections.Generic;
using Assent;
using Typescriptr;
using Xunit;

namespace Typescript.Tests.CollectionTypes
{
    public class CollectionGeneratorTests
    {
        class TypeWithArrayProp
        {
            public string[] ArrayProp { get; set; }
        }
        
        [Fact]
        public void Generator_TypeWithArrayProperty_ShouldRenderToArray()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithArrayProp)});

            this.Assent(generated.Types);
        }

        class TypeWithEnumerableProp
        {
            public IEnumerable<int> EnumerableProp { get; set; }
        }

        [Fact]
        public void Generator_TypeWithEnumerableProp_ShouldRenderToArray()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithEnumerableProp)});

            this.Assent(generated.Types);
        }

        class IsEnumerable : List<int> {}
        
        class TypeWithPropThatInheritsFromIEnumerable
        {
            public IsEnumerable DerivedFromList { get; set; }
        }

        [Fact]
        public void Generator_TypeWithPropThatDerivesFromEnumerable_ShouldRenderToArray()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithPropThatInheritsFromIEnumerable)});

            this.Assent(generated.Types);
        }
    }
}