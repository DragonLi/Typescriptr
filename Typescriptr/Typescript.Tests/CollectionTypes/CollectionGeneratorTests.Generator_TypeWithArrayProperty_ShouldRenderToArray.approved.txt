declare namespace Api {
  interface TypeWithArrayProp {
    arrayProp: string[];
  }
}
